include/boost/url/bnf/parse.hpp

```diff
@@ -77,214 +96,176 @@ parse(
-   number defined in @ref boost::urls::error.
+   @code
+   using bnf::parse;
+   if (bnf::parse(str, ec, bnf_element_1, bnf_element_2, bnf_element_3)) {
+     std::cout << "String " << str << " parsed successfully\n";
+   }
+   @endcode
 
    @par Exception Safety
-     Calls to `parse` overloads may throw, although they should
-     prefer setting @ref ec instead.
 
-   @see @ref boost::urls::error
+     Defined by the types of `args`.
 
    @return `true` if the string matches the BNF elements successfully and all
-   chars are consumed, `false` otherwise
+   chars are consumed
 
-   @param[in] s The input string this function should parse
+   @param s The input string
 
-   @param[out] ec The error code this function should set if it fails
+   @param ec Set to the error, if any occurred. If the string is not completely consumed,
+   but the beginning of the string matches the elements, `ec` is set to @ref error::leftover.
 
-   @param[out] t0 T0 1st BNF element
+   @param args BNF elements
 
-   @param[out] t1 T1 2nd BNF element
+   @see
+       @ref error.
 
-   @param[out] tn ...Tn [3rd, 4th, ...] BNF elements
+   @par References
+   @li [Backus–Naur form](https://en.wikipedia.org/wiki/Backus%E2%80%93Naur_form)
 
 */
 template<
+#ifdef BOOST_URL_DOCS
+    class... Args
+#else
     class T0,
-    class... Tn>
+    class... Tn
+#endif
+>
 bool
 parse_string(
     string_view s,
     error_code& ec,
+#ifdef BOOST_URL_DOCS
+    Args&&... args
+#else
     T0&& t0,
-    Tn&&... tn);
+    Tn&&... tn
+#endif
+);
 
-/** @brief Parse a complete string and throw on failure
+/** Parse a complete string into BNF elements and throw on failure
 
-   This function parses a complete string @ref s into the Backus–Naur form (BNF)
-   elements [@ref t0, @ref t1, ...]. If the string is not completely consumed,
+   This function parses a complete string `s` into the Backus–Naur form (BNF)
+   elements in `args`. If the string is not completely consumed,
    an error is thrown.
 
-   The function returns `true`when the input is parsed successfully into the
-   elements [@ref t0, @ref t1, ...]. If the input cannot be parsed into the
-   BNF elements, the function throw a @ref system_error.
+   @par Example
+
+   @code
+   try {
+      using bnf::parse;
+      bnf::parse(str, ec, bnf_element_1, bnf_element_2, bnf_element_3);
+   } catch (boost::urls::system_error& e) {
+      std::cout << e.what() << '\n';
+   }
+   @endcode
 
    @par Exception Safety
+
      Exceptions thrown on invalid input.
 
-   @see @ref boost::urls::error
+   @param s The input string
 
-   @param[in] s The input string this function should parse
+   @param args BNF elements
 
-   @param[out] t0 T0 1st BNF element
+   @throw system_error on failure
 
-   @param[out] t1 T1 2nd BNF element
+   @see
+       @ref error.
 
-   @param[out] tn ...Tn [3rd, 4th, ...] BNF elements
+   @par References
+   @li [Backus–Naur form](https://en.wikipedia.org/wiki/Backus%E2%80%93Naur_form)
 
```

Comment:
The wikipedia entry for Backus-Naur form in no way helps explain what `parse` does so why link it?

Reply:
Because it's what is being parsed. But that's already been removed for another reason anyway.